checkov/common/util/secrets.py

```diff
@@ -146,13 +146,15 @@ def omit_secret_value_from_checks(check: BaseCheck, check_result: dict[str, Chec
     censored_code_lines = []
 
     if CheckCategories.SECRETS in check.categories and check_result.get('result') == CheckResult.FAILED:
-        secrets.update([str(secret) for key, secret in entity_config.items() if key.startswith(f'{check.id}_secret')])
+        secrets.update([str(secret) for key, secret in entity_config.items() if
+                        key.startswith(f'{check.id}_secret')])
 
     if resource_attributes_to_omit and check.entity_type in resource_attributes_to_omit and \
-            resource_attributes_to_omit.get(check.entity_type) in entity_config:
-        secret = entity_config.get(resource_attributes_to_omit.get(check.entity_type, ''), [])
-        if isinstance(secret, list) and secret:
-            secrets.add(secret[0])
+            [attribute for attribute in resource_attributes_to_omit.get(check.entity_type) if attribute in entity_config]:  # type:ignore[union-attr]
```

Comment:
```suggestion
```
you can just remove this line. the for loop will do the same thing anyway. if there is nothing to iterate over then the inner block will be skipped.

Reply:
you're right